### ISSUE TITLE
Command got signal SEGV.

### ISSUE THREAD

**Issue by mimoo** [zshipko/ocaml-rs#118]
I'm getting the error in the title at the moment, and I see that I'm the first one talking about that in the issues :D hopefully I'll find some solution that will help the next guy.

**Comment by mimoo** [comment]
I'm wondering if it could be that I implement `FromValue` on references:

```rust
        unsafe impl<'a> ocaml::FromValue<'a> for &$name {
            fn from_value(value: ocaml::Value) -> Self {
                let x: ocaml::Pointer<Self> = ocaml::FromValue::from_value(value);
                x.as_ref()
            }
        }
```

or if in the last stable version I still need to implement a finalizer to call drop

**Comment by mimoo** [comment]
OK so I isolated the exported Rust function that was crashing (using lldb), and I changed the parameter from `&MyOpaqueType` to `ocaml::Pointer<MyOpaqueType>` and it works. I guess I need to implement `FromValue` on `T` and not `&T`

**Comment by mimoo** [comment]
it's kind of weird because to book actually works with references: https://zshipko.github.io/ocaml-rs/02_type_conversion.html (and it doesn't even have to implement `FromValue`, I'm guessing the last beta version implements it for you with `custom!`?)

**Comment by mimoo** [comment]
the examples in the [tests](https://github.com/zshipko/ocaml-rs/blob/v0.22.4/test/src/custom.rs) seem to only use `ocaml::Pointer<_>` as arguments of functions as well

**Comment by zshipko** [comment]
Yeah, in the latest version `FromValue` is implemented for `&T` and `&mut T` where `T: Custom`: https://github.com/zshipko/ocaml-rs/blob/cae584cfd7bab89ac8634f6b50acd27191f8b815/src/custom.rs#L195 

They use `Pointer` to do the conversion.

**Comment by mimoo** [comment]
do you know why I was getting this errors with that implementation?

```rust
unsafe impl<'a> ocaml::FromValue<'a> for &$name {
            fn from_value(value: ocaml::Value) -> Self {
                let x: ocaml::Pointer<Self> = ocaml::FromValue::from_value(value);
                x.as_ref()
            }
        }
```

in any case now things work by using `ocaml::Pointer<_>` everywhere, but I would have expected that to work : o
